sdk/core/core-http/src/policies/exponentialRetryPolicy.ts

```diff
@@ -139,6 +140,15 @@ async function retry(
 ): Promise<HttpOperationResponse> {
   function shouldPolicyRetry(responseParam?: HttpOperationResponse): boolean {
     const statusCode = responseParam?.status;
+    if (
+      statusCode &&
+      statusCode === 503 &&
+      response &&
+      response.headers.get(Constants.HeaderConstants.RETRY_AFTER)
+    ) {
+      return false;
+    }
```

Comment:
I decided to put this in a separate conditional to make it easier to think through.